src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationAttribute.kt

```diff
@@ -182,29 +180,42 @@ class LocationAttribute private constructor(
     }
   }
 
+  fun isAvailableFor(court: Court, startingOnDateTime: LocalDateTime): AvailabilityStatus = check(court, startingOnDateTime)
+
   private fun check(court: Court, startingOnDateTime: LocalDateTime): AvailabilityStatus {
-    if (locationStatus == LocationStatus.INACTIVE) {
-      return AvailabilityStatus.NONE
-    }
+    if (locationStatus == LocationStatus.INACTIVE) return AvailabilityStatus.NONE
 
     return when (locationUsage) {
       LocationUsage.SHARED -> AvailabilityStatus.SHARED
-      LocationUsage.COURT -> {
-        if (allowedParties.isNullOrBlank()) {
-          AvailabilityStatus.COURT_ANY
-        } else {
-          if (isPartyAllowed(court.code)) {
-            AvailabilityStatus.COURT_ROOM
-          } else {
-            AvailabilityStatus.NONE
-          }
-        }
+      LocationUsage.COURT -> when {
+        allowedParties.isNullOrBlank() -> AvailabilityStatus.COURT_ANY
+        isPartyAllowed(court.code) -> AvailabilityStatus.COURT_ROOM
+        else -> AvailabilityStatus.NONE
       }
-      LocationUsage.SCHEDULE -> TODO()
+
+      LocationUsage.SCHEDULE -> getScheduleAvailability(court, startingOnDateTime)
       else -> return AvailabilityStatus.NONE
     }
   }
 
+  /**
+   * We need to look at the schedules as a whole to determine availability. A schedule on its own is not enough.
+   */
+  private fun getScheduleAvailability(court: Court, dateAndTime: LocalDateTime): AvailabilityStatus {
```

Comment:
This was the TODO part.  The logic is the same (but for courts) regardless of single room or multiple room model.